### ISSUE TITLE
Couple of questions

### ISSUE THREAD

**Issue by tarensanders** [danilodpsantos/gt3x2csv#17]
Hi Danilo et al,

I came across this package and it looks like it could be a lifesaver. We have >1200 gt3x files to convert to run in GGIR, which would take about 42ish hours on Actilife.
I'm trying to use this package to save some time, but I keep running into issues. I'm hoping you could point me in the right direction. I'm also happy to contribute to the package, but fair warning that my R is pretty scrappy compared to my Python.

Here are my questions:
1. When I run the following:
`gt3x_2_csv("~/Documents/Test Data", outdir = "~/Documents/Test Output", verbose = TRUE)`
I get the following error: 
```
TRACE [2020-04-08 11:35:39] Initializing runtime counter.
TRACE [2020-04-08 11:35:39] Processing files sequentially.
Error in { : task 1 failed - "argument is of length zero"
```

Oddly, adding an argument for the log file seems to stop that error, but I could not see why from the code.

2. When I try and run the code on a single file without parallel I get a crash. From the log file it looks like this occurring when it is trying to append the accelerometer data to the header csv. The crash has the output in [this screenshot](https://drive.google.com/file/d/1b34lhvMaUcu4i7bVBvXvCQPNQ9EsxumO/view?usp=sharing). 

3. When I run in parallel, I get the error `Error in unserialize(socklist[[n]]) : error reading from connection`. I know that's not helpful. It does look like it's at the same place as the other errors: when trying to join the header and the accel data. And ideas?

4. I've opened a PR to fix a very minor mistake in the logging. It was confusing me when I was trying to run on just 4 cores (#16 )

5. I'm curious about these lines 
```R
as.data.table(.[, -4]) %>%
transmute(`Accelerometer X` = Y %>% as.character,
                   `Accelerometer Y` = X %>% as.character,
                  `Accelerometer Z` = Z %>% as.character)
```
Is this a mistake? Are X and Y being flipped?

**Comment by danilodpsantos** [comment]
Hello Taren, Thanks for reaching!

Your points and your collaboration are more than welcome. 
I am an R beginner and frankly, I'm learning along the process.
We've designed this package for our use in a project. We've had to convert 10.000 files, you can imagine how long it would take.

I'll answer question 5, since we'll have to take a closer look at the issues you reported in points 1 to 3.

X and Y being flipped was actually a correction of a mistake we've found in the read.gt3x package, that we use in this package to read the log.bin files. 
In the official Actigraph site, they explicitly define that the axis order in the log.bin file's columns is Y, X and Z. In the read.gt3x function though, they use the order X, Y, Z which seems to be wrongly specified.

I'll take a look at the issues you've reported and talk to the other collaborators find an answer. I am sorry I can't give you a deadline for that though.

Regards,
Danilo

**Comment by tarensanders** [comment]
Hey Danilo,

Thanks for getting back to me. Really appreciate your help.

Looks like you're right about read.gt3x - the swapped X/Y is the first issue on their repo. Interesting that it hasn't been fixed.

After looking into this a bit more, my two guesses are either memory allocation or a package incompatibility. Is there any chance you could run `devtools::session_info()` and post the result (after loading gt3x2csv)? I'll see if I can recreate your environment with the same package versions to see if that solves the issue.

FYI, my session info is:

```
─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       macOS Catalina 10.15.6      
 system   x86_64, darwin17.0          
 ui       RStudio                     
 language (EN)                        
 collate  en_AU.UTF-8                 
 ctype    en_AU.UTF-8                 
 tz       Australia/Sydney            
 date     2020-08-07                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version    date       lib source                                  
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 4.0.0)                          
 backports     1.1.8      2020-06-17 [1] CRAN (R 4.0.0)                          
 bit           4.0.3      2020-07-30 [1] CRAN (R 4.0.2)                          
 bit64         4.0.2      2020-07-30 [1] CRAN (R 4.0.2)                          
 callr         3.4.3      2020-03-28 [1] CRAN (R 4.0.0)                          
 cli           2.0.2      2020-02-28 [1] CRAN (R 4.0.0)                          
 codetools     0.2-16     2018-12-24 [1] CRAN (R 4.0.2)                          
 crayon        1.3.4      2017-09-16 [1] CRAN (R 4.0.0)                          
 data.table    1.13.0     2020-07-24 [1] CRAN (R 4.0.2)                          
 desc          1.2.0      2018-05-01 [1] CRAN (R 4.0.0)                          
 devtools      2.3.1      2020-07-21 [1] CRAN (R 4.0.2)                          
 digest        0.6.25     2020-02-23 [1] CRAN (R 4.0.0)                          
 doSNOW        1.0.18     2019-07-27 [1] CRAN (R 4.0.2)                          
 dplyr         1.0.1      2020-07-31 [1] CRAN (R 4.0.2)                          
 ellipsis      0.3.1      2020-05-15 [1] CRAN (R 4.0.0)                          
 fansi         0.4.1      2020-01-08 [1] CRAN (R 4.0.0)                          
 foreach       1.5.0      2020-03-30 [1] CRAN (R 4.0.2)                          
 fs            1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                          
 gdata         2.18.0     2017-06-06 [1] CRAN (R 4.0.2)                          
 generics      0.0.2      2018-11-29 [1] CRAN (R 4.0.0)                          
 glue          1.4.1      2020-05-13 [1] CRAN (R 4.0.0)                          
 gt3x2csv    * 0.1.1      2020-08-04 [1] Github (danilodpsantos/gt3x2csv@7348ce1)
 gtools        3.8.2      2020-03-31 [1] CRAN (R 4.0.2)                          
 hms           0.5.3      2020-01-08 [1] CRAN (R 4.0.0)                          
 iterators     1.0.12     2019-07-26 [1] CRAN (R 4.0.2)                          
 lifecycle     0.2.0      2020-03-06 [1] CRAN (R 4.0.0)                          
 logger        0.1        2019-01-02 [1] CRAN (R 4.0.2)                          
 magrittr      1.5        2014-11-22 [1] CRAN (R 4.0.0)                          
 memoise       1.1.0      2017-04-21 [1] CRAN (R 4.0.2)                          
 pillar        1.4.6      2020-07-10 [1] CRAN (R 4.0.2)                          
 pkgbuild      1.1.0      2020-07-13 [1] CRAN (R 4.0.2)                          
 pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.0.0)                          
 pkgload       1.1.0      2020-05-29 [1] CRAN (R 4.0.0)                          
 prettyunits   1.1.1      2020-01-24 [1] CRAN (R 4.0.0)                          
 processx      3.4.3      2020-07-05 [1] CRAN (R 4.0.0)                          
 ps            1.3.3      2020-05-08 [1] CRAN (R 4.0.0)                          
 purrr         0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                          
 R6            2.4.1      2019-11-12 [1] CRAN (R 4.0.0)                          
 Rcpp          1.0.5      2020-07-06 [1] CRAN (R 4.0.0)                          
 read.gt3x     0.1.0.9000 2020-08-04 [1] Github (THLfi/read.gt3x@c4b83cc)        
 remotes       2.2.0      2020-07-21 [1] CRAN (R 4.0.2)                          
 rlang         0.4.7      2020-07-09 [1] CRAN (R 4.0.2)                          
 rprojroot     1.3-2      2018-01-03 [1] CRAN (R 4.0.0)                          
 rstudioapi    0.11       2020-02-07 [1] CRAN (R 4.0.0)                          
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 4.0.2)                          
 snow          0.4-3      2018-09-14 [1] CRAN (R 4.0.2)                          
 stringi       1.4.6      2020-02-17 [1] CRAN (R 4.0.0)                          
 stringr       1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                          
 testthat      2.3.2      2020-03-02 [1] CRAN (R 4.0.0)                          
 tibble        3.0.3      2020-07-10 [1] CRAN (R 4.0.2)                          
 tidyr         1.1.1      2020-07-31 [1] CRAN (R 4.0.2)                          
 tidyselect    1.1.0      2020-05-11 [1] CRAN (R 4.0.0)                          
 usethis       1.6.1      2020-04-29 [1] CRAN (R 4.0.2)                          
 vctrs         0.3.2      2020-07-15 [1] CRAN (R 4.0.2)                          
 vroom         1.2.1      2020-05-12 [1] CRAN (R 4.0.2)                          
 withr         2.2.0      2020-04-20 [1] CRAN (R 4.0.0)       
```

Cheers,
Taren

**Comment by danilodpsantos** [comment]
Hi Taren,

My guests are the same as yours. 
I have completed the file conversion a couple of months ago, and since then I didn't work in the package anymore.

It probably has to do with R or package versions, since it's been quite some time.

My session info as you asked:

```R

- Session info -------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.6.3 (2020-02-29)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  Portuguese_Brazil.1252      
 ctype    Portuguese_Brazil.1252      
 tz       America/Sao_Paulo           
 date     2020-08-16                  

- Packages -----------------------------------------------------------------------------------
 package     * version  date       lib source          
 assertthat    0.2.1    2019-03-21 [1] CRAN (R 3.6.3)  
 backports     1.1.5    2019-10-02 [1] CRAN (R 3.6.1)  
 bit           1.1-15.2 2020-02-10 [1] CRAN (R 3.6.2)  
 bit64         0.9-7    2017-05-08 [1] CRAN (R 3.6.2)  
 callr         3.4.3    2020-03-28 [1] CRAN (R 3.6.3)  
 cli           2.0.2    2020-02-28 [1] CRAN (R 3.6.3)  
 codetools     0.2-16   2018-12-24 [2] CRAN (R 3.6.3)  
 crayon        1.3.4    2017-09-16 [1] CRAN (R 3.6.3)  
 data.table    1.12.8   2019-12-09 [1] CRAN (R 3.6.3)  
 desc          1.2.0    2018-05-01 [1] CRAN (R 3.6.3)  
 devtools      2.2.2    2020-02-17 [1] CRAN (R 3.6.3)  
 digest        0.6.25   2020-02-23 [1] CRAN (R 3.6.3)  
 doSNOW        1.0.18   2019-07-27 [1] CRAN (R 3.6.3)  
 dplyr         0.8.5    2020-03-07 [1] CRAN (R 3.6.3)  
 ellipsis      0.3.0    2019-09-20 [1] CRAN (R 3.6.3)  
 fansi         0.4.1    2020-01-08 [1] CRAN (R 3.6.3)  
 foreach       1.5.0    2020-03-30 [1] CRAN (R 3.6.3)  
 fs            1.4.0    2020-03-31 [1] CRAN (R 3.6.2)  
 gdata         2.18.0   2017-06-06 [1] CRAN (R 3.6.3)  
 glue          1.4.0    2020-04-03 [1] CRAN (R 3.6.3)  
 gt3x2csv    * 0.1.1    2020-04-22 [1] local           
 gtools        3.8.2    2020-03-31 [1] CRAN (R 3.6.3)  
 hms           0.5.3    2020-01-08 [1] CRAN (R 3.6.3)  
 iterators     1.0.12   2019-07-26 [1] CRAN (R 3.6.3)  
 lifecycle     0.2.0    2020-03-06 [1] CRAN (R 3.6.3)  
 logger        0.1      2019-01-02 [1] CRAN (R 3.6.3)  
 magrittr      1.5      2014-11-22 [1] CRAN (R 3.6.3)  
 memoise       1.1.0    2017-04-21 [1] CRAN (R 3.6.3)  
 pillar        1.4.3    2019-12-20 [1] CRAN (R 3.6.3)  
 pkgbuild      1.0.6    2019-10-09 [1] CRAN (R 3.6.3)  
 pkgconfig     2.0.3    2019-09-22 [1] CRAN (R 3.6.3)  
 pkgload       1.0.2    2018-10-29 [1] CRAN (R 3.6.3)  
 prettyunits   1.1.1    2020-01-24 [1] CRAN (R 3.6.3)  
 processx      3.4.2    2020-02-09 [1] CRAN (R 3.6.3)  
 ps            1.3.2    2020-02-13 [1] CRAN (R 3.6.3)  
 purrr         0.3.3    2019-10-18 [1] CRAN (R 3.6.3)  
 R6            2.4.1    2019-11-12 [1] CRAN (R 3.6.3)  
 Rcpp          1.0.4.6  2020-04-09 [1] CRAN (R 3.6.3)  
 read.gt3x     0.1.0    2020-04-22 [1] git2r (@8d8b816)
 remotes       2.1.1    2020-02-15 [1] CRAN (R 3.6.3)  
 rlang         0.4.5    2020-03-01 [1] CRAN (R 3.6.3)  
 rprojroot     1.3-2    2018-01-03 [1] CRAN (R 3.6.3)  
 rstudioapi    0.11     2020-02-07 [1] CRAN (R 3.6.3)  
 sessioninfo   1.1.1    2018-11-05 [1] CRAN (R 3.6.3)  
 snow          0.4-3    2018-09-14 [1] CRAN (R 3.6.1)  
 stringi       1.4.6    2020-02-17 [1] CRAN (R 3.6.2)  
 stringr       1.4.0    2019-02-10 [1] CRAN (R 3.6.3)  
 testthat      2.3.2    2020-03-02 [1] CRAN (R 3.6.3)  
 tibble        3.0.0    2020-03-30 [1] CRAN (R 3.6.3)  
 tidyr         1.0.2    2020-01-24 [1] CRAN (R 3.6.3)  
 tidyselect    1.0.0    2020-01-27 [1] CRAN (R 3.6.3)  
 usethis       1.5.1    2019-07-04 [1] CRAN (R 3.6.3)  
 vctrs         0.2.4    2020-03-10 [1] CRAN (R 3.6.3)  
 vroom         1.2.0    2020-01-13 [1] CRAN (R 3.6.3)  
 withr         2.1.2    2018-03-15 [1] CRAN (R 3.6.3)  

```

I Believe that the best solution would be to actually update the package.

Thanks for reaching,
Danilo

**Comment by tarensanders** [comment]
Hey Danilo,

Thanks for this. It looks like there are pretty substantial differences to the `read.gt3x` package between our two environment: https://github.com/THLfi/read.gt3x/compare/8d8b816..c4b83cc

I'll try reverting to the version you have and see if that solves it. Don't stress about updating the package. We're about to start on a pooled analysis (we'll have data from lots of teams), so I'm happy to contribute to get it working. It'll be super useful for us.

Cheers,
Taren

**Comment by angelorbastos** [comment]
Hey Taren,

So we finally had some time to sit down and take a look at your first three questions, and it all seemed to stem from a simple error with defaulting the `logfile` parameter to `NULL`.

We pushed a fix today to the `dev` branch, so when you have some time could you check if it now works for you? That way we can move it to `master`.

Thanks for your feedback and collaboration on the package! That's always very welcome.

Best regards,
Angelo Rosa

**Comment by tarensanders** [comment]
Hi Angelo,

Really appreciate you guys looking into this. Unfortunately the `dev` build still gives me errors although it does look like your commit fixes the logfile issue.

I continuously get a C stack usage error. I just spent a bit of time trying to debug it, and I found that I can run `header_csv` fine, but it is `save_accel` that causes an error (see [screenshot](https://drive.google.com/file/d/13PAWLcyWsAFNAbhGNSXhFUDB03HOWAtl/view?usp=sharing)). 

It seems like the error is caused by `vroom_write`. My understanding is that `0\` is a reserved character in ASCII, and so it is getting confused when trying to write this. To debug, I set a breakpoint at the error and stepped through. The `accel_df` looks correct to me, and weirdly in writes correctly if I step through the `vroom_write` function, but not if I just let the function run.

I've tried reverting my version of `vroom` down to 1.2 but I'm still getting the error. I'll keep trying and see what I can work out.

Thanks again guys, you've done a great job on this package. Not sure why it's not playing ball for me.

**Comment by tarensanders** [comment]
Ah I solved it!

The issue is in writing the strings from the data.table. So I've moved these to doubles and now it runs perfectly. I was able to get it to run and produce the right output on my machine. It's super fast too, you guys did an amazing job (this is so much better than Actilife).

I just open #18 with the changes.

**Comment by danilodpsantos** [comment]
Hi Taren, thanks for noticing this new issue and fixing it!

The former version of the package used fwrite function to append the acceleration data to the header, we've changed the function to vroom_write but forgot to change the strings to doubles. We'll review the package as a whole in the next days.

I'm really happy to hear that the package will be useful for you! I hope it will save a lot of time and make data conversion less burdensome.

If you find anything else that might be subject to improvements we are happy to hear

**Comment by tarensanders** [comment]
Hi Danilo,

Glad I could help in some small way. You guys did a great job with this package. I just ran it across 20 week-long files and got:

`TRACE [2020-29-10 09:04:01] Total elapsed time: 25.21 seconds.`

Phenomenal, that probably would have taken 10+ minutes in Actilife.

I think we can close this issue now.
